### ISSUE TITLE
Extend PyG dataloader support

### ISSUE THREAD

**Issue by arnaudvl** [mims-harvard/TDC#76]
It would be great if closer integration with PyTorch Geometric could be provided for the various TDC datasets. While it is straightforward to convert e.g. SMILES into the graph format for single instances (not sure however how to interpret the node features or include edge features and labels?) via `tdc.chem_utils.MolConvert`, it is not clear to me how this can be turned into a PyTorch Geometric compatible dataset/dataloader including the desired features. E.g. the OGB repository provides support as follows:

```python
from ogb.graphproppred import PygGraphPropPredDataset

dataset = PygGraphPropPredDataset(name='ogbg-molhiv' )
split_idx = dataset.get_idx_split()
batch_size = 32
dl_tr = DataLoader(dataset[split_idx['train']], batch_size=batch_size, shuffle=True)
dl_val = DataLoader(dataset[split_idx['valid']], batch_size=batch_size, shuffle=False)
dl_te = DataLoader(dataset[split_idx['test']], batch_size=batch_size, shuffle=False)
```
Maybe I am missing something here and the above is already possible via the otherwise great TDC library?

**Comment by kexinhuang12345** [comment]
hey, thanks for the suggestion! I believe we indeed have some basic node/edge attributes (atom/bond index, atom charge/chirality, etc) supported in the `MolConvert` class (https://github.com/mims-harvard/TDC/blob/afe08ca727cda11b3a65af5caadf16f8352878a9/tdc/chem_utils.py#L3449). To clarify, are you saying we should provide more options for the atom/bond features?

**Comment by kexinhuang12345** [comment]
Note that you can also do something like this:
```
from tdc.single_pred import ADME
data = ADME(name = 'Caco2_Wang', convert_format = 'ECFP4')
```
It will create a separate column in the dataframe with the converted format.

However, as our current dataloader in default returns pandas dataframe, some of the convert format wouldn't work, for example, the DGL graph object would return error when we put it in a dataframe. So, in these cases, it would be ideal to simply return the DGL/PyG objects. We would work to integrate this feature in the future.
